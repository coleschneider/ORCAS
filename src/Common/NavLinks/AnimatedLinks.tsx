import * as React from 'react';
import posed from 'react-pose';


export const AnimatedNav = posed.nav({
  closed: {
    delay: 100,
    paddingTop: 0,
    y: '-85%',
  },
  open: {
    beforeChildren: true,
    delayChildren: 60,
    paddingTop: 78,
    staggerChildren: 125,
    y: '0%',
  },
});

export const Item = posed.li({
  closed: { y: 20, opacity: 0 },
  open: {
    // delayChildren: 50,
    beforeChildren: true,
    opacity: 1,
    // staggerChildren: 0,
    y: 0,
  },
});
export const SubItem = posed.div({
  closed: { height: 0, },
  open: { height: 'auto' },
});