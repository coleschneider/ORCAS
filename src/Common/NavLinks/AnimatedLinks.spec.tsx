import * as React from 'react';
import * as AnimatedLinks from './AnimatedLinks';

describe('AnimatedLinks', () => {
  describe('AnimatedNav', () => {
    it('should match the screenshot when the prop is open', () => {
      const wrapper = shallow(<AnimatedLinks.AnimatedNav posed="open" />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should match the screenshot when the prop is closed', () => {
      const wrapper = shallow(<AnimatedLinks.AnimatedNav posed="closed" />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Item', () => {
    it('should match the screenshot when the props is closed', () => {
      const wrapper = shallow(<AnimatedLinks.Item posed="open" />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should match the screenshot when the props is open', () => {
      const wrapper = shallow(<AnimatedLinks.Item posed="closed" />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
