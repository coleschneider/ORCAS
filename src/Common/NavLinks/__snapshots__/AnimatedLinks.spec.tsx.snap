// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedLinks AnimatedNav should match the screenshot when the prop is closed 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`AnimatedLinks AnimatedNav should match the screenshot when the prop is open 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`AnimatedLinks Item should match the screenshot when the props is closed 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`AnimatedLinks Item should match the screenshot when the props is open 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
